Department, warning, and general sign code fix. And brand new custom plaques!

uomo91 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Department and warning sign code was bad. This makes it less bad.

Use a pen on any existing sign, or craft a sign backing with plastic then put it on a wall and use a pen on it. Tah dah! You can make brand new signs, or change existing ones! Except it's crappy. The old code was a giant unmaintainable switch that also excluded a bunch of existing sign types. This fixes that, sorts sign types in a sane manner in the code, and establishes a variable for them being named categorically and alphabetically in a tidy list when you use a pen on any sign, and some other things including a bunch of consistency tweaks and fixes.

This also adds a new customizable plaque to the game. Stackcraft it using gold, then use a fancy fountain pen to engrave it. You can hand it to somebody, mount it on a wall, and wrench it off a wall. Once engraved, it cannot be changed again. (it can, however, be stolen or destroyed)

To do:
Figure out how armor types and the other flags work and if I should change them, remove them, whatever.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

![penmenu](https://user-images.githubusercontent.com/51800976/79701111-06285800-8260-11ea-8ece-72ce98868131.png)

![rotate](https://user-images.githubusercontent.com/51800976/80431262-9268f980-88b6-11ea-9669-03c7d7f8498b.png)


Signs are fun.
Rebuild round start signs when a department goes kaboom and things get repaired, change all of security's signs to electric shock warning signs to protest their liberal use of stunbatons, put biohazard signs around all the toilets, add a variety of useful or fun signs (many previously unable to make through gameplay means) to your new or remodeled room, the possibilities are ~~LITERALLY LIMITLESS~~ pretty limited but way better than they used to be.

![plaque](https://user-images.githubusercontent.com/51800976/80431273-99900780-88b6-11ea-834c-e14b1149b872.png)

Plaques are even more fun.
Honor the science department with a plaque commemorating their having completed the station's goal, or engineering for saving the station from a delamination, or the paramedics for saving somebody's life.
Or just steal the Captain's pen to make an insulting plaque and hang it outside his quarters.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: You can now craft plaques using gold in hand, and with a fountain pen can engrave a title and description on them. They can be mounted on walls. No good deed shall ever go forgotten now.
fix: Refactored signs heavily. A bunch of sign types previously left out of using a pen on a sign to change it are now included. They're also sorted sensibly now. You can no longer change sign types that shouldn't be changeable (guidance signs, maps, etc)
fix: Signs (and plaques) can now be repaired with a welder.
fix: Signs placed on walls are now only placed on one side of that wall, and are not visible through the wall.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

uomo91 committed:
Oh God I hope this works...

uomo91 committed:
It didn't work.

uomo91 committed:
Making things better.

uomo91 committed:
Making things even better.

uomo91 committed:
That made things worse.

uomo91 committed:
Sorting works! \o/

uomo91 committed:
Tidying some thing ups.

uomo91 committed:
Adding a custom plaque to the game.

uomo91 committed:
Some plaque tweaks.

uomo91 committed:
Makes plaques more durable than signs.

uomo91 committed:
Adds the stack crafting.

uomo91 committed:
Plaques take a little time to engrave, signs can't be changed instantly anymore.

bobbahbrown commented on code/game/objects/structures/signs/signs_departments.dm:
```diff
@@ -1,96 +1,164 @@
+	sign_change_name = "Department - Medbay Alt"
+	desc = "The Intergalactic symbol of Medical institutions. You'll probably get help here."
+	icon_state = "bluecross2"
+	is_editable = TRUE
```
make a base department sign/sign that has is_editable as TRUE as you seem to be setting it on every subtype of departments

uomo91 replied:
As per our discussion elsewhere, there doesn't seem to be a better way to do it offhand.

uomo91 committed:
Comments, and renamed the global list to avoid confusion.

uomo91 committed:
Appends "sign" to every name, makes department signs not all caps, fixes some names and descriptions.

uomo91 committed:
Touching up plaque customization and display, removing one last all caps.

uomo91 committed:
Plaques can now be engraved in hand too.

uomo91 committed:
Newline.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+	var/plaque_path = /obj/structure/customplaque //This points the item to make the proper structure when placed on a wall.
+	var/unengraved = TRUE
+
+/obj/structure/customplaque/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
```
```suggestion
/obj/structure/customplaque/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = NONE)
```
`NONE` is a define for no flags

uomo91 replied:
Fixed.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+	switch(damage_type)
+		if(BRUTE)
+			if(damage_amount)
+				playsound(src.loc, 'sound/weapons/smash.ogg', 80, TRUE)
```
```suggestion
				playsound(loc, 'sound/weapons/smash.ogg', 80, TRUE)
```
`src` refers to the thing calling the proc. All variables without a `.` access and not in the call of the proc are implied to have `src.`, which makes it not necessary to add it.

Only add `.src` when the object variable has the same name as the proc call variable.
For example, if `/obj/structure/customplaque` has an object variable called `damage_amount` while `play_attack_sound` also has a variable of the same name.
In such case, in order to differentiate them you'd do `src.damage_amount` for the object variable and just `damage_amount` for the proc variable.

uomo91 replied:
Fixed.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+
+
+/obj/structure/customplaque/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_WRENCH)
+		user.visible_message("<span class='notice'>[user] starts removing [src]...</span>", \
+							 "<span class='notice'>You start unfastening [src].</span>")
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 40))
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			user.visible_message("<span class='notice'>[user] unfastens [src].</span>", \
+								 "<span class='notice'>You unfasten [src].</span>")
+			var/obj/item/customplaque/CP = new (get_turf(user))
+			CP.name = name //Copy over the plaque structure variables to the plaque item we're creating when we unwrench it.
+			CP.desc = desc
+			CP.unengraved = unengraved
+			CP.setDir(dir)
+			qdel(src) //The plaque structure on the wall goes poof and only the plaque item from unwrenching remains.
+		return
```
Check for examples of `wrench_act` and move everything inside the `if(I.tool_behaviour == TOOL_WRENCH)` into a ` /obj/structure/customplaque/wrench_act(mob/living/user, obj/item/wrench/W)` proc.

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+			desc = "The plaque reads: '[descriptionchoice]'"
+			unengraved = FALSE //The plaque now has a name, description, and can't be altered again.
+			user.visible_message("<span class='notice'>[user] engraves [src].</span>", \
+								 "<span class='notice'>You engrave [src].</span>")
+	else if(istype(I, /obj/item/pen) && unengraved)
+		to_chat(user, "<span class='warning'>Your pen isn't fancy enough to engrave this! Find a fountain pen.</span>") //Go steal the Curator's.
+	else
+		return ..()
```
Instead of doing
```nix
if
  stuff
else
  stuff
else
  return ..()
```
do:
```nix
if
  stuff
  return
if
  stuff
  return
return ..()
```

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+/obj/item/customplaque/attack_self(mob/user)
+	. = ..()
+	setDir(turn(dir, 90))
+	
```
```suggestion

```

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+
+/obj/item/customplaque/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(isturf(target) && proximity)
```
Do an early return.

uomo91 replied:
Sure thing.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+			desc = "The plaque reads: '[descriptionchoice]'"
+			unengraved = FALSE
+			user.visible_message("<span class='notice'>[user] engraves [src].</span>", \
+								 "<span class='notice'>You engrave [src].</span>")
+	else if(istype(I, /obj/item/pen) && unengraved)
```
Same here, do `if-return-if` instead of `if-else`
Also end this with `return ..()`

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -6,8 +6,10 @@
 	max_integrity = 100
 	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
-	var/buildable_sign = 1 //unwrenchable and modifiable
+	var/buildable_sign = 1 //unwrenchable
```
```suggestion
	var/buildable_sign = TRUE //unwrenchable
```
This is a boolean, so it should be signalized as such.

uomo91 replied:
Fixed.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -40,87 +52,48 @@
-		qdel(src)
+		user.visible_message("<span class='notice'>[user] begins changing [src].</span>", \
+							 "<span class='notice'>You begin changing [src].</span>")
+		if(do_after(user, 40, target = src)) //Small delay for changing signs instead of it being instant, so somebody could be shoved or stunned to prevent them from doing so.
```
Do an early return

uomo91 replied:
Done.

uomo91 committed:
Update code/game/objects/structures/customplaque.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

uomo91 committed:
Update code/game/objects/structures/customplaque.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

uomo91 committed:
Mirroring some fixes from customplaque.dm

trollbreeder commented on code/game/objects/structures/signs/signs_warning.dm:
```diff
@@ -1,122 +1,181 @@
-	desc = "A sign that warns of high-power testing equipment in the area. That's either a really powerful laser... or a satellite landing on some person's head."
+	name = "\improper TESTING AREA sign"
+	sign_change_name = "Warning - Testing Area"
+	desc = "A sign that warns of high-power testing equipment in the area."
```
Aww, why'd you remove the soul from this sign's description?

uomo91 replied:
This is primarily used for rooms with the BEPIS and experimenter in it, now. It seemed like outdated flavortext, but I suppose it could be given new flavortext.

trollbreeder reviewed (commented):
Grammar review time! For most review comments there's suggested changes, so you can easily correct grammatical errors. Or you can do whatever.

uomo91 committed:
A bunch of stuff.

Iamgoofball commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,146 @@
+			else
+				playsound(loc, 'sound/weapons/tap.ogg', 50, TRUE)
+		if(BURN)
+			playsound(loc, 'sound/items/welder.ogg', 80, TRUE)
```
is this not handled elsewhere already?

uomo91 replied:
I have absolutely no idea.

Qustinnus replied:
it is.

uomo91 replied:
I can just remove the proc completely then?

uomo91 replied:
I believe I fixed this. Signs now make a slash noise when hit but I got rid of the rest of the code there.

uomo91 replied:
I just axed the slash sound completely, it's kind of out of place for a plastic sign.

Iamgoofball commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,146 @@
+		CP.obj_integrity = obj_integrity
+		CP.setDir(dir)
+		qdel(src) //The plaque structure on the wall goes poof and only the plaque item from unwrenching remains.
+	return TRUE
```
. = TRUE

Rohesie replied:
Why would you not `return TRUE` at the end of the proc?

uomo91 replied:
Per a conversation, this seems resolved.

Iamgoofball commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,146 @@
+	return TRUE
+
+/obj/structure/customplaque/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/pen/fountain) && unengraved)
```
check ungraved first, istype is more complex, easiest checks always go first to save time

uomo91 replied:
Done.

Iamgoofball commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -4,21 +4,25 @@
 
 /obj/structure/sign/basic
-	name = "blank sign"
-	desc = "How can signs be real if our eyes aren't real? Use a pen to change the decal."
```
cant believe you did this great meme dirty like this

uomo91 replied:
Sacrifices must be made.

Iamgoofball replied:
no

Iamgoofball commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +34,118 @@
 	user.examinate(src)
 
+GLOBAL_VAR(editable_sign_types)
+/proc/populate_editable_sign_types() //The first time a pen is used on any sign, this populates the above, a global list of all the signs that you can set a sign backing to with a pen.
```
code doc this

Iamgoofball replied:
also this should probably be done in a NO_FIRE subsystem

uomo91 replied:
For now I documented it, and will figure out the subsystem thing later. Thanks.

uomo91 replied:
Opted to just move it to _globalvars and make it GLOBAL_LIST_EMPTY

Iamgoofball commented on code/game/objects/structures/signs/signs_warning.dm:
```diff
@@ -1,122 +1,181 @@
 	desc = "A warning sign which reads 'DISPOSAL: LEADS TO MORGUE'."
 	icon_state = "bodysposal"
-
-/obj/structure/sign/warning/pods
```
you axed this by accident

uomo91 replied:
It's still there, it just got moved. A bunch of these types got moved around to categorize them more clearly.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,146 @@
+	max_integrity = 200 //Twice as durable as regular signs.
+	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
+	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
+	var/unengraved = TRUE //Custom plaque structures and items both start "unengraved", once engraved with a fountain pen their text can't be altered again.
```
```suggestion
	///Custom plaque structures and items both start "unengraved", once engraved with a fountain pen their text can't be altered again.
	var/unengraved = TRUE
```

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,146 @@
+	custom_materials = list(/datum/material/gold = 2000)
+	max_integrity = 200
+	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
+	var/plaque_path = /obj/structure/customplaque //This points the item to make the proper structure when placed on a wall.
+	var/unengraved = TRUE
```
```suggestion
	///This points the item to make the proper structure when placed on a wall.
	var/plaque_path = /obj/structure/customplaque 
	///Custom plaque structures and items both start "unengraved", once engraved with a fountain pen their text can't be altered again.
	var/unengraved = TRUE
```

uomo91 replied:
Done.

uomo91 committed:
Unwrenched signs now lay horizontal.

uomo91 commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,104 @@
+	max_integrity = 200
+	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
+	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
+	var/unengraved = TRUE
```
Done!

uomo91 commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,104 @@
+	w_class = WEIGHT_CLASS_NORMAL
+	custom_materials = list(/datum/material/gold = 2000)
+	var/plaque_path = /obj/structure/customplaque
+	var/unengraved = TRUE
```
Done!

uomo91 commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+	icon = 'icons/obj/decals.dmi'
+	icon_state = "blankplaque"
+	name = "blank plaque"
+	desc = "A blank plaque. Use a fancy pen to engrave it, use a wrench to detatch it from the wall."
```
Fixed.

uomo91 commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+	icon = 'icons/obj/decals.dmi'
+	icon_state = "blankplaque"
+	name = "blank plaque"
+	desc = "A blank plaque. Place it on a wall and use a fancy pen to engrave it, use a wrench to detatch it from the wall."
```
Fixed.

uomo91 commented on code/game/objects/structures/customplaque.dm:
```diff
@@ -0,0 +1,123 @@
+							 "<span class='notice'>You begin engraving [src].</span>")
+		if(do_after(user, 40, target = src)) //This spits out a visible message that somebody is engraving a plaque, then has a delay.
+			if(!Adjacent(user)) //Make sure user is adjacent still
+				to_chat(user, "<span class='warning'>You have to stand next to the plaque to engrave it!</span>")
```
Done.

uomo91 committed:
A lot more stuff, turned a negative variable into a positive.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +36,133 @@
+	S.obj_integrity = obj_integrity
+	S.setDir(dir)
+	qdel(src)
+	return ..()
```
```suggestion

```
Parent is already called here.

uomo91 replied:
I removed it.

Qustinnus commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +36,133 @@
-								 "<span class='notice'>You unfasten [src].</span>")
-			var/obj/item/sign_backing/SB = new (get_turf(user))
+///This is a global list of all signs you can change an existing sign or new sign backing to, when using a pen on them.
+GLOBAL_VAR(editable_sign_types)
```
put this in _globalvars and use GLOBAL_LIST_EMPTY

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +36,133 @@
+		if(!choice)
+			return
+		user.visible_message("<span class='notice'>You begin changing [src].</span>")
+		if(!do_after(user, 40, target = src))
```
```suggestion
		if(!do_after(user, 4 SECONDS, target = src))
```

uomo91 replied:
Done.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +36,133 @@
-		//It's import to clone the pixel layout information
+		user.visible_message("<span class='notice'>[user] begins changing [src].</span>", \
+							 "<span class='notice'>You begin changing [src].</span>")
+		if(!do_after(user, 40, target = src)) //Small delay for changing signs instead of it being instant, so somebody could be shoved or stunned to prevent them from doing so.
```
```suggestion
		if(!do_after(user, 4 SECONDS, target = src)) //Small delay for changing signs instead of it being instant, so somebody could be shoved or stunned to prevent them from doing so.
```

uomo91 replied:
Also done.

uomo91 committed:
Fix chat inconsistency, remove deprecated and unused variable.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -30,111 +36,133 @@
+	user.visible_message("<span class='notice'>[user] starts removing [src]...</span>", \
+						 "<span class='notice'>You start unfastening [src].</span>")
+	I.play_tool_sound(src)
+	if(I.use_tool(src, user, 40))
```
Do an early return here.

uomo91 replied:
Done.

uomo91 replied:
OK, that causes a crash.

uomo91 replied:
I figured it out, done.

uomo91 committed:
Apparently buildable_sign is used elsewhere, whoops.

uomo91 committed:
Adding documentation to this, fixing a bool being 0 in another file.

uomo91 committed:
Don't know why this was here, sign dir is 100% irrelevant.

uomo91 committed:
Fixes some sound malarkey.

uomo91 committed:
Changes delays to be readable, removes unneeded return.

uomo91 committed:
Fixes these two early returns, it needed return TRUE at the end to work.

uomo91 committed:
Trying to fix a revert I messed up...

uomo91 committed:
Moving plaques to a directory, moving old sign plaque types into it. Map changes.

uomo91 committed:
Rename since I relocated this object entirely.

uomo91 committed:
Signs (and plaques) now properly place, and aren't visible through walls.

uomo91 commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -4,137 +4,177 @@
-		qdel(src)
+	if(!iswallturf(target) || !proximity)
+		return
+	var/turf/T = target
+	var/turf/userT = get_turf(user)
+	var/obj/structure/sign/S = new sign_path(userT) //We place the sign on the turf the user is standing, and pixel shift it to the target wall, as below.
+	//This is to mimic how signs and other wall objects are usually placed by mappers, and so they're only visible from one side of a wall.
+	var/dir = get_dir(userT, T)
+	switch(dir)
+		if(NORTH)
+			S.pixel_y = 32
+		if(SOUTH)
+			S.pixel_y = -32
+		if(EAST)
+			S.pixel_x = 32
+		if(WEST)
+			S.pixel_x = -32
+		else //Signs cannot be placed diagonally.
+			to_chat(user, "<span class='warning'>Your reach is unsteady, stand directly in front of the wall you wish to place a sign on")
+			qdel(S)
+			return 
+	user.visible_message("<span class='notice'>[user] fastens [src] to [T].</span>", \
+						 "<span class='notice'>You attach the sign to [T].</span>")
+	playsound(T, 'sound/items/deconstruct.ogg', 50, TRUE)
+	S.obj_integrity = obj_integrity
+	S.setDir(dir)
+	qdel(src)
```
If somebody knows a better way to do this, PLEASE tell me. Issue is placing signs diagonally and easily figuring out what their pixel shift should be.

uomo91 committed:
Comment to help people in the future.

uomo91 committed:
Signs and plaques can be placed diagonally now.

uomo91 committed:
Removes duplicated code line.

uomo91 commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,171 @@
+	var/obj/structure/plaque/S = new plaque_path(userT) //We place the plaque on the turf the user is standing, and pixel shift it to the target wall, as below.
+	//This is to mimic how signs and other wall objects are usually placed by mappers, and so they're only visible from one side of a wall.
+	var/dir = get_dir(userT, T)
+	switch(dir)
+		if(NORTH)
+			S.pixel_y = 32
+		if(SOUTH)
+			S.pixel_y = -32
+		if(EAST)
+			S.pixel_x = 32
+		if(WEST)
+			S.pixel_x = -32
+		if(NORTHEAST)
+			S.pixel_y = 32
+			S.pixel_x = 32
+		if(NORTHWEST)
+			S.pixel_y = 32
+			S.pixel_x = -32
+		if(SOUTHEAST)
+			S.pixel_y = -32
+			S.pixel_x = 32
+		if(SOUTHWEST)
+			S.pixel_y = -32
+			S.pixel_x = -32
```
If there is a shorter, cleaner way to accomplish this, PLEASE let me know.

Rohesie replied:
```cp
if(dir & NORTH)
	S.pixel_y = 32
else if(dir & SOUTH)
	S.pixel_y = -32
if(dir & EAST)
	S.pixel_x = 32
else if(dir & WEST)
	S.pixel_x = -32
```

Rohesie replied:
The short version is still preferable, as the performance gain from the other is very minor, and this is not a hot proc.

uomo91 replied:
OK, I changed it.

uomo91 committed:
Blank signs now commonly spawn in maint, blank plaques uncommonly.

uomo91 committed:
Repairing signs & plaques with a welder is now good to go.

uomo91 committed:
Moves the GLOB, makes it start as an empty list.

Rohesie commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,207 @@
+	name = "blank plaque"
+	desc = "A blank plaque, use a fancy pen to engrave it. It can be detatched from the wall with a wrench."
+	anchored = TRUE
+	opacity = 0
```
```suggestion
	opacity = FALSE
```

Rohesie commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,207 @@
+	S.setDir(dir)
+	qdel(src)
+
+/obj/item/plaque/Move(atom/new_loc, direct = 0)
+	// Pulling, throwing, or conveying a plaque does not rotate it.
+	var/old_dir = dir
+	. = ..()
+	setDir(old_dir)
+
+/obj/item/plaque/attack_self(mob/user)
+	. = ..()
+	setDir(turn(dir, 90))
```
Unless changing direction matters for something, feel free to remove both of these.

uomo91 replied:
I was considering that yeah, just wasn't sure. Will do.

uomo91 replied:
Done.

uomo91 committed:
Update code/game/objects/structures/plaques/_plaques.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>

uomo91 committed:
Gets rid of some oldcode.

tralezab commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,196 @@
+		return TRUE
+	user.visible_message("<span class='notice'>[user] starts repairing [src]...</span>", \
+						 "<span class='notice'>You start repairing [src].</span>")
+	if(!I.use_tool(src, user, 4 SECONDS, volume =50 ))
```
```suggestion
	if(!I.use_tool(src, user, 4 SECONDS, volume = 50))
```

uomo91 replied:
Done, thanks!

tralezab commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,196 @@
+		return TRUE
+	user.visible_message("<span class='notice'>[user] starts repairing [src]...</span>", \
+						 "<span class='notice'>You start repairing [src].</span>")
+	if(!I.use_tool(src, user, 4 SECONDS, volume =50 ))
```
```suggestion
	if(!I.use_tool(src, user, 4 SECONDS, volume = 50))
```

tralezab reviewed (approved):
other than minor stuff that don't matter (but here you go) much this is good to go imo

uomo91 committed:
Apply suggestions from code review

Co-Authored-By: tralezab <40974010+tralezab@users.noreply.github.com>

uomo91 committed:
Simplifying dir switch to two if elses.

uomo91 committed:
Plaques take one sheet of gold to graft instead of five.

uomo91 committed:
Just making this wording in crafting a little clearer.

Rohesie commented on code/game/objects/structures/plaques/_plaques.dm:
```diff
@@ -0,0 +1,183 @@
+	. = ..()
+	if(!iswallturf(target) || !proximity)
+		return
+	var/turf/T = target
+	var/turf/userT = get_turf(user)
+	var/obj/structure/plaque/S = new plaque_path(userT) //We place the plaque on the turf the user is standing, and pixel shift it to the target wall, as below.
```
```suggestion
	var/turf/target_turf = target
	var/turf/user_turf = get_turf(user)
	var/obj/structure/plaque/placed_plaque = new plaque_path(userT) //We place the plaque on the turf the user is standing, and pixel shift it to the target wall, as below.
```

uomo91 replied:
Done.

uomo91 committed:
They can be removed with wrenches, so I don't think this is proper.

uomo91 committed:
Makes variables descriptive.

uomo91 committed:
More descriptive variables, a little dmdoc, move things around to be next to like things.

uomo91 committed:
Remove a redundant definition.

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -72,7 +70,7 @@
 						 "<span class='notice'>You unfasten [src].</span>")
 	var/obj/item/sign_backing/unwrenched_sign = new (get_turf(user))
-	if(type != /obj/structure/sign/basic) //If it's still just a basic sign backing, we can (and should) skip some of the below variable transfers.
+	if(type != /obj/structure/sign/) //If it's still just a basic sign backing, we can (and should) skip some of the below variable transfers.
```
```suggestion
	if(type != /obj/structure/sign) //If it's still just a basic sign backing, we can (and should) skip some of the below variable transfers.
```

Rohesie commented on code/game/objects/structures/signs/_signs.dm:
```diff
@@ -32,7 +30,7 @@
 	max_integrity = 100
 	///The type of sign structure that will be created when placed on a turf, the default looks just like a sign backing item.
-	var/sign_path = /obj/structure/sign/basic
+	var/sign_path = /obj/structure/sign/
```
```suggestion
	var/sign_path = /obj/structure/sign
```

uomo91 replied:
Thanks.

uomo91 committed:
Apply suggestions from code review

Co-Authored-By: Rohesie <rohesie@gmail.com>

uomo91 committed:
Fixes mapping path, rename sign_backing to just sign.
